src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/MidEventCallback.java

```diff
@@ -90,6 +90,19 @@ public JsonNode invoke(String caseTypeId,
         return dataJsonNode(content.getData());
     }
 
+    private void removeNextPageFieldData(CaseDetails currentCaseDetails, Integer order, String caseTypeId, String eventId) {
+        if (order != null) {
+            Set<String> wizardPageFields = uiDefinitionRepository
+                .getWizardPageCollection(caseTypeId, eventId)
+                .stream()
+                .filter(wizardPage -> wizardPage.getOrder() > order)
+                .map(WizardPage::getWizardPageFieldNames)
+                .flatMap(Set::stream)
+                .collect(Collectors.toSet());
+            currentCaseDetails.getData().entrySet().removeIf(entry -> wizardPageFields.contains(entry.getKey()));
+        }
+    }
```

Comment:
this method probably does not belong to the MidEventCallback. Could we introduce an utility class if we don't have a suitable one already? suggestions: `WizardPagesUtils`